core/container/externalbuilder/copy_test.go

```diff
@@ -45,51 +58,77 @@ var _ = Describe("copy", func() {
 		os.RemoveAll(destRootDir)
 	})
 
-	When("dest dir does not exist", func() {
-		BeforeEach(func() {
-			destRootDir, err = ioutil.TempDir("", "dest-")
-			Expect(err).NotTo(HaveOccurred())
-			err = os.RemoveAll(destRootDir)
+	It("Copies all files and subdirectories", func() {
+		err = CopyDir(logger, srcRootDir, destRootDir)
+		Expect(err).NotTo(HaveOccurred())
+
+		_, err := os.Stat(destRootDir)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = os.Stat(filepath.Join(destRootDir, "file-in-root.txt"))
+		Expect(err).NotTo(HaveOccurred())
```

Comment:
Is this trying to ensure the file exists? `Expect(...).To(BeARegularFile())`? I think most of the places you're doing the `os.Stat` to get a side-effect of an error should be replaced with a [matcher](https://onsi.github.io/gomega/#working-with-files).

I'd also expect some assertions on file contents to ensure we didn't just do the `os.Create` and forget the copy. (Because it sure looks like I can comment out the `io.Copy` and all of the tests will still pass...